pkg/storage/unified/resource/storage_backend.go

```diff
@@ -0,0 +1,777 @@
+package resource
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand/v2"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/bwmarrin/snowflake"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/storage/unified/resourcepb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	defaultListBufferSize = 100
+)
+
+// Unified storage backend based on KV storage.
+type kvStorageBackend struct {
+	snowflake  *snowflake.Node
+	kv         KV
+	dataStore  *dataStore
+	metaStore  *metadataStore
+	eventStore *eventStore
+	notifier   *notifier
+	builder    DocumentBuilder
+}
+
+var _ StorageBackend = &kvStorageBackend{}
+
+func NewkvStorageBackend(kv KV) *kvStorageBackend {
```

Comment:
This should still be called `NewKvStorageBackend`, even if `k` is not upper-case in the type name.